workflow/rules/mapping.smk

```diff
@@ -214,3 +214,126 @@ rule apply_bqsr:
         java_opts="",  # optional
     wrapper:
         "v2.3.2/bio/gatk/applybqsr"
+
+
+rule map_reads_vg_giraffe:
+    input:
+        reads=get_map_reads_input,
+        idx="resources/pangenome/hprc-v1.0-mc-grch38.xg",
+    output:
+        "results/mapped/vg/{sample}_mapped.bam",
+    log:
+        "logs/mapped/vg/{sample}.log",
+    benchmark:
+        "benchmarks/vg_giraffe/{sample}.tsv"
+    conda:
+        "../envs/vg.yaml"
+    threads: 40
+    params:
+        lambda wc, input: " -f ".join(input.reads),  # potential issue: in case of single end reads, get map_reads_input() returns a string and join() could create a problem.
+    shell:
+        "vg giraffe -x {input.idx} -f {params} --output-format BAM -t {threads}  > {output} 2> {log}"
+
+
+rule sort_mapped_vg:
+    input:
+        "results/mapped/vg/{sample}_mapped.bam",
+    output:
+        "results/mapped/vg/{sample}_sorted.bam",
+    log:
+        "logs/samtools_sort_vg/{sample}.log",
+    threads: 8
+    wrapper:
+        "v2.3.2/bio/samtools/sort"
+
+
+# keep only primary chromosomes
+# use -f 2 to keep only properly paired reads, the mate of a read that's on a nonprimary chromosome is problematic for AddOrReplaceReadGroups, because we remove
+# all the other nonprimary chromosome from the header.
+
+
+rule keep_only_primary_chr:
+    input:
+        "results/mapped/vg/{sample}_sorted.bam",
+        "results/mapped/vg/{sample}_sorted.bai",
+    output:
+        bam="results/mapped/vg/{sample}_extracted.bam",
+        idx="results/mapped/vg/{sample}_extracted.bai",
+    log:
+        "logs/samtools_view_primary_chr/{sample}.log",
+    benchmark:
+        "benchmarks/samtools_view_primary_chr/{sample}.tsv"
+    params:
+        region="GRCh38.chr1 GRCh38.chr2 GRCh38.chr3 GRCh38.chr4 GRCh38.chr5 GRCh38.chr6 GRCh38.chr7 GRCh38.chr8 GRCh38.chr9 GRCh38.chr10 GRCh38.chr11 GRCh38.chr12 GRCh38.chr13 GRCh38.chr14 GRCh38.chr15 GRCh38.chr16 GRCh38.chr17 GRCh38.chr18 GRCh38.chr19 GRCh38.chr20 GRCh38.chr21 GRCh38.chr22 GRCh38.chrX GRCh38.chrY GRCh38.chrM",
```

Comment:
This should not be hardcoded. Workflow has to work for all species in principle (as long as there is a pangenome index). We have a selection for the first n chromosomes already implemented in the workflow (at the level of the variants). Isn't that enough?

Reply:
Do you refer to a function that looks in the header, finds the names of first n chromosomes from `config['ref']['n_chromosomes']`  and that is then used here to extract only records belonging to them? Wouldn't that be the only way other than hardcoding them? As far as I checked, that kind of implementation doesn't exist in the workflow, and the vg index is in `xg` format that cannot be easily subset like a fasta. Can you eloborate further on how we can achieve what you mean?